src/main/java/io/nats/client/impl/OrderedManager.java

```diff
@@ -14,73 +14,66 @@
+        super(conn, js, stream, so, serverCC, queueMode, dispatcher);
         expectedConsumerSeq = 1; // always starts at 1
     }
 
     @Override
-    boolean manage(Message msg) {
-        if (msg != null) {
-            long receivedConsumerSeq = msg.metaData().consumerSequence();
-            if (expectedConsumerSeq != receivedConsumerSeq) {
-                try {
-                    expectedConsumerSeq = 1; // consumer always starts with consumer sequence 1
+    protected boolean subManage(Message msg) {
+        long receivedConsumerSeq = msg.metaData().consumerSequence();
+        if (expectedConsumerSeq != receivedConsumerSeq) {
+            handleErrorCondition();
+            return true;
+        }
+        expectedConsumerSeq++;
+        return false;
+    }
+
+    @Override
+    protected void handleHeartbeatError() {
+        handleErrorCondition();
+    }
+
+    private void handleErrorCondition() {
+        try {
+            expectedConsumerSeq = 1; // consumer always starts with consumer sequence 1
 
-                    // 1. shutdown the managers, for instance stops heartbeat timers
-                    for (MessageManager mm : sub.managers) {
-                        mm.shutdown();
-                    }
+            // 1. shutdown the manager, for instance stops heartbeat timers
+            sub.manager.shutdown();
 
-                    // 2. re-subscribe. This means kill the sub then make a new one
-                    //    New sub needs a new deliver subject
-                    String newDeliverSubject = sub.connection.createInbox();
-                    sub.reSubscribe(newDeliverSubject);
+            // 2. re-subscribe. This means kill the sub then make a new one
+            //    New sub needs a new deliverSubject
+            String newDeliverSubject = sub.connection.createInbox();
+            sub.reSubscribe(newDeliverSubject);
 
-                    // 3. make a new consumer using the same deliver subject but
-                    //    with a new starting point
-                    ConsumerConfiguration userCC = ConsumerConfiguration.builder(serverCC)
-                        .deliverPolicy(DeliverPolicy.ByStartSequence)
-                        .deliverSubject(newDeliverSubject)
-                        .startSequence(lastStreamSeq + 1)
-                        .startTime(null) // clear start time in case it was originally set
-                        .build();
-                    js._createConsumerUnsubscribeOnException(stream, userCC, sub);
+            // 3. make a new consumer using the same deliver subject but
+            //    with a new starting point
+            ConsumerConfiguration userCC = ConsumerConfiguration.builder(serverCC)
+                .deliverPolicy(DeliverPolicy.ByStartSequence)
+                .deliverSubject(newDeliverSubject)
+                .startSequence(Math.max(1, lastStreamSeq + 1))
```

Comment:
This ensures handling of an error condition that occurs before any messages are received.